pkg/apis/pipeline/pod/template.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pod
+
+import (
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// PodTemplate holds pod specific configuration
+// +k8s:deepcopy-gen=true
+type Template struct {
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	// +optional
+	NodeSelector map[string]string `yaml:"nodeSelector,omitempty" json:"nodeSelector,omitempty"`
```

Comment:
@vdemeester i had to add yaml tags on the `Template` struct to be able to deserialize it. But the yaml tags were not present on the other structs, is this normal ? Am i using the wrong serialization tool ?

Reply:
:thinking: They shouldn't be needed indeed…

Reply:
if i don't put them `NodeSelector` is expected to be a property named `nodeselector` by `yaml.Unmarshal` rather than `nodeSelector`

Reply:
:thinking: We should use kubernetes's unmarshal mechanism then (I think they do expose it from somewhere)

Reply:
i thought it was something like this, i'll change that

Reply:
Looks like you're using `ghodss/yaml` to unmarshal.
I changed the code to use it and removed the yaml tags on the `Template` struct.

The `ghodss/yaml` you use is in vendor and does not support UnmarshalStrict, i commented out some tests that checked unmarshaling errors. (these are new tests i added)